rsplan/primitives.py

```diff
@@ -307,9 +309,9 @@ def _interpolated(
         # straight segments) and dtheta (step size / turn radius) for curved segments.
         step = step_size if self.is_straight else step_size / self.turn_radius
 
-        seg_pts = np.arange(0, magnitude, step)
-
-        # Add segment endpoint if the list of segment points is not empty
-        seg_pts = np.append(seg_pts, [magnitude]) if seg_pts.any() else np.array([0.0])
+        # Since np.arrange does not handle floating point arithmetic well, we use
```

Comment:
Maybe something like `Prefer linspace to arange to avoid floating point errors. Will distribute remainder distance across steps instead of having a short final step.`

Reply:
sgtm